File: hdd/two_sum/two_sum.kt
// 다트와 마찬가지로
// Set을 이용해 시간 절약
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {

        var answer = IntArray(2)
        var sets = nums.toSet()

        for(i : Int in 0 .. nums.size - 1){

            if(sets.contains(target - nums.get(i)))   {
                val index = nums.indexOf(target - nums.get(i))
                if(index != -1 && index != i){
                    answer.set(0,i)
                    answer.set(1,index)
                    break
                }
            }
        }
        return answer
    }
}

File: hdd/two_sum/two_sum.dart
//contains의 시간복잡도가 1이 Set을 만들어
//List에서 불필요한 반복을 돌 필요 없도록 변경

class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Set<int> numsSet = nums.toSet();
    for (int i = 0; i < nums.length; i++) {
      if (numsSet.contains(target - nums[i])) {
        var result = nums.indexWhere((e) => e == target - nums[i], i + 1);
        if (result != -1) {
          return [i, result];
        }
      }
    }
    return [];
  }
}
